packages/react-baseline-inputs/src/FieldSet.tsx

```diff
@@ -62,9 +65,19 @@ export const FieldSet: React.FC<FieldSetProps> = ({
     <fieldset
       id={id}
       aria-describedby={error ? errorLabelId : undefined}
+      aria-required={required ? "true" : undefined}
       className={join(fieldSetClassNames)}
     >
-      {legend && <legend className={join(legendClassNames)}>{legend}</legend>}
+      {legend && (
+        <legend className={join(legendClassNames)}>
+          {legend}
+          {required && (
+            <span className={join(requiredIndicatorClassNames)}>
+              {requiredIndicator || "*"}
```

Comment:
Rather than using `||`, you should default `requiredIndicator = "*"` above.

If someone didn't want the required indicator to show, they might pass `false` or `null`. In that scenario, this would still show `*` because `(null || "*") == "*"`.

Reply:
Also, is the wrapping span going to be a problem here?

```jsx
requiredIndicator={<span>required/span>}
```

This would result in: 

```
<span><span>required</span></span>
```

Reply:
ah good call.

Reply:
Nested spans is not _necessarily_ a problem. If you had a use case where you needed the extra markup, that may be the desired result

Reply:
> Also, is the wrapping span going to be a problem here?

Hmm, this poses an interesting issue — `<legend>` only accepts phrasing content as valid elements, which puts a container `<div>` out of the question.

I could do something like:
```
requiredIndicator || <span className={join(requiredIndicatorClassNames)}>*</span>
```

but that would mean you'd need to byo markup and class names when replacing the default.

Would expecting a string instead of a ReactNode in this instance be acceptable?

Reply:
On second thought I think it's more flexible to keep it as is — @andrewcroce's right that nested spans aren't necessarily a problem — it's valid html (just not a very nice markup pattern)

Reply:
I could imagine wanting to put an icon or something in here, so I think you'd need a node. I'd probably say we can't make sure everyone does everything right all the time, and its kind of up to the engineer to make sure their markup is valid. 

Sometimes escape hatches just lead to another trap. It's a risk we might need to allow.

Reply:
Another possible solution would be a render prop:

```jsx
renderRequired={({ className }) => <span className={className}>*</span>}
```

This gives you the benefit of being able to pass down some info (such as className)